subsys/net/ip/net_context.c

```diff
@@ -1357,12 +1357,14 @@ static int context_setup_udp_packet(struct net_context *context,
 		return ret;
 	}
 
-	ret = net_udp_create(pkt,
-			     net_sin((struct sockaddr *)
-				     &context->local)->sin_port,
-			     dst_port);
-	if (ret) {
-		return ret;
+	if (net_context_get_type(context) != SOCK_RAW) {
+		ret = net_udp_create(pkt,
+				     net_sin((struct sockaddr *)
+					     &context->local)->sin_port,
+				     dst_port);
+		if (ret) {
+			return ret;
+		}
 	}
 
 	ret = context_write_data(pkt, buf, len, msg);
```

Comment:
put this on a else after previous if()

Reply:
Sorry, I don't understand how you want this to be ordered, could you post the code you'd like to see?

Reply:
I believe @tbursztyka meant something like this. As there are now multiple `if(ret) { ... }` checks in this function, you could remove one of them. So the code could look like this:
```
	if (net_context_get_type(context) != SOCK_RAW) {
		ret = net_udp_create(pkt,
				     net_sin((struct sockaddr *)
					     &context->local)->sin_port,
				     dst_port);
	}

	if (ret == 0) {
		ret = context_write_data(pkt, buf, len, msg);
	}

	return ret;
}
```

Reply:
Alright, personally I think that just makes the diff noisier, but I've done it.